File: backend/ws_wa_store_ccadmin/src/main/java/com/ccadmin/app/sale/model/entity/StoreVirtualConfigEntity.java
package com.ccadmin.app.sale.model.entity;

import com.ccadmin.app.shared.model.entity.AuditTableEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.io.Serializable;
import java.math.BigDecimal;

@Entity
@Table(name = "store_virtual_config")
public class StoreVirtualConfigEntity extends AuditTableEntity implements Serializable {

    @Id
    public String StoreCod;
    public String AllowsAutomaticDelivery = "N";
    public BigDecimal AutomaticDeliveryRadiusKm;
    public String AllowsScheduledDelivery = "N";
    public BigDecimal ScheduledDeliveryMaxRadiusKm;
    public String AllowsStorePickup = "N";
    public int PreparationTimeMinutes;

    public StoreVirtualConfigEntity() {
    }
}

File: frontend/web_ccadmin_lt/src/app/enterprise/sale/model/entity/StoreVirtualConfigEntity.ts
import { AuditTableEntity } from "src/app/enterprise/shared/model/entity/AuditTableEntity";

export class StoreVirtualConfigEntity extends AuditTableEntity
{
    public StoreCod: string = "";
    public AllowsAutomaticDelivery: string = "N";
    public AutomaticDeliveryRadiusKm: number | null = null;
    public AllowsScheduledDelivery: string = "N";
    public ScheduledDeliveryMaxRadiusKm: number | null = null;
    public AllowsStorePickup: string = "N";
    public PreparationTimeMinutes: number = 0;

    public constructor()
    {
        super();
    }
}
